js/validation/index.js

```diff
@@ -12,16 +12,179 @@
+    "review": validateReview,
+
+    "softwareSuggestions": noValidation, // TODO: validate SoftwareSourceCode
+    "maintainer": validateActors,
+    "contIntegration": validateUrls,
+    "continuousIntegration": validateUrls,
+    "buildInstructions": validateUrls,
+    "developmentStatus": validateText, // TODO: use only repostatus strings?
+    "embargoDate": validateDate,
+    "embargoEndDate": validateDate,
+    "funding": validateText,
+    "issueTracker": validateUrls,
+    "referencePublication": noValidation, // TODO?
+    "readme": validateUrls,
+};
+
+const creativeWorkFieldValidators = {
+    "@id": validateUrl,
+    "id": validateUrl,
+
+    "author": validateActors,
+    "citation": validateCreativeWorks, // TODO
+    "contributor": validateActors,
+    "copyrightHolder": validateActors,
+    "copyrightYear": validateNumbers,
+    "creator": validateActors, // TODO: still in codemeta 2.0, but removed from master
+    "dateCreated": validateDate,
+    "dateModified": validateDate,
+    "datePublished": validateDate,
+    "editor": validatePersons,
+    "encoding": noValidation,
+    "funder": validateActors, // TODO: may be other types
+    "keywords": validateTexts,
+    "license": validateCreativeWorks,
+    "producer": validateActors,
+    "provider": validateActors,
+    "publisher": validateActors,
+    "sponsor": validateActors,
+    "version": validateNumberOrText,
+    "isAccessibleForFree": validateBoolean,
+    "isPartOf": validateCreativeWorks,
+    "hasPart": validateCreativeWorks,
+    "position": noValidation,
+    "identifier": noValidation, // TODO
+    "description": validateText,
+    "name": validateText,
+    "sameAs": validateUrls,
+    "url": validateUrls,
+};
+
+const roleFieldValidators = {
+    "roleName": validateText,
+    "startDate": validateDate,
+    "endDate": validateDate,
+
+    "schema:author": validateActor
+};
+
+const personFieldValidators = {
+    "@id": validateUrl,
+    "id": validateUrl,
+
+    "givenName": validateText,
+    "familyName": validateText,
+    "email": validateText,
+    "affiliation": validateOrganizations,
+    "identifier": validateUrls,
+    "name": validateText,  // TODO: this is technically valid, but should be allowed here?
+    "url": validateUrls,
+};
+
+const organizationFieldValidators = {
+    "@id": validateUrl,
+    "id": validateUrl,
+
+    "email": validateText,
+    "identifier": validateUrls,
+    "name": validateText,
+    "address": validateText,
+    "sponsor": validateActors,
+    "funder": validateActors, // TODO: may be other types
+    "isPartOf": validateOrganizations,
+    "url": validateUrls,
+
+    // TODO: add more?
+};
+
+const reviewFieldValidators = {
+    "reviewAspect": validateText,
+    "reviewBody": validateText,
+}
+
+function switchCodemetaContext(codemetaJSON, contextUrl) {
+    const previousCodemetaContext = codemetaJSON["@context"];
+    codemetaJSON["@context"] = contextUrl;
+    return previousCodemetaContext;
+}
```

Comment:
Instead of altering the argument and returning the previous context; could you return a new object with the new context, and not change the argument? I think it would be less confusing, and removes the need to switch it back.